model/task_lifecycle.go

```diff
@@ -1209,25 +1213,37 @@ func getBuildStatus(buildTasks []task.Task) buildStatus {
 		return buildStatus{status: evergreen.BuildCreated, allTasksBlocked: allTasksBlocked}
 	}
 
+	var hasUnfinishedTask bool
+	var unfinishedEssentialTasks []string
 	// Check if tasks are started but not finished.
 	for _, t := range buildTasks {
 		if t.Status == evergreen.TaskStarted {
-			return buildStatus{status: evergreen.BuildStarted}
+			hasUnfinishedTask = true
 		}
 		if t.Activated && !t.Blocked() && !t.IsFinished() {
```

Comment:
You use this logic and the status logic above to; would it be helpful to move this to a WillRun helper function or something?